### PR TITLE
feat(intersection): add intersection type

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #48** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/48" target="_blank">Graphite</a>  👈
    * **PR #49** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/49" target="_blank">Graphite</a> 
      * **PR #50** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/50" target="_blank">Graphite</a> 
        * **PR #51** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/51" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/48?utm_source=stack-comment).

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/48?utm_source=stack-comment).


Job Reference: G9qW3Gh7LcbjHSuwHmyz

**Pull request merged**
